src/Analyzers/CSharp/CodeFixes/UseImplicitOrExplicitType/UseExplicitTypeCodeFixProvider.cs

```diff
@@ -74,42 +80,69 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
             if (declarationContext is VariableDeclarationSyntax varDecl)
             {
                 typeSyntax = varDecl.Type;
+                declarationSyntax = varDecl.Variables.Single().Identifier.Parent;
             }
             else if (declarationContext is ForEachStatementSyntax forEach)
             {
                 typeSyntax = forEach.Type;
+                declarationSyntax = forEach.Identifier.Parent;
             }
             else if (declarationContext is DeclarationExpressionSyntax declarationExpression)
             {
                 typeSyntax = declarationExpression.Type;
-                if (declarationExpression.Designation.IsKind(SyntaxKind.ParenthesizedVariableDesignation, out ParenthesizedVariableDesignationSyntax variableDesignation))
+
+                if (declarationExpression.Designation.IsKind(SyntaxKind.ParenthesizedVariableDesignation, out ParenthesizedVariableDesignationSyntax? variableDesignation))
                 {
                     parensDesignation = variableDesignation;
                 }
             }
             else
             {
-                throw ExceptionUtilities.UnexpectedValue(declarationContext.Kind());
+                throw ExceptionUtilities.UnexpectedValue(declarationContext?.Kind());
             }
 
             if (parensDesignation is null)
             {
                 typeSyntax = typeSyntax.StripRefIfNeeded();
 
+                var newTypeSymbol = semanticModel.GetTypeInfo(typeSyntax, cancellationToken).ConvertedType;
+                RoslynDebug.AssertNotNull(newTypeSymbol);
+
+                if (newTypeSymbol.NullableAnnotation == NullableAnnotation.Annotated && declarationSyntax is not null)
+                {
+                    var syntaxFacts = document.GetRequiredLanguageService<ISyntaxFactsService>();
+
+                    // It's possible that the var shouldn't be annotated nullable, check assignments to the variable and 
+                    // determine if it needs to be null
+                    var encapsulatingNode = syntaxFacts.GetIOperationRootNode(node);
+                    Contract.ThrowIfNull(encapsulatingNode);
+
+                    var operationScope = semanticModel.GetRequiredOperation(encapsulatingNode, cancellationToken);
```

Comment:
throw if this is null as well?

Reply:
Won't `GetRequiredOperation` throw for that case?

Reply:
yup.  i didn't notice this was GetRequiredOperation.  THanks.